services/mirror/mirror_pull.go

```diff
@@ -371,13 +364,31 @@ func runSync(ctx context.Context, m *repo_model.Mirror) ([]*mirrorSyncResult, bo
 		log.Trace("SyncMirrors [repo: %-v Wiki]: running git remote update...", m.Repo)
 		stderrBuilder.Reset()
 		stdoutBuilder.Reset()
-		if err := git.NewCommand("remote", "update", "--prune").AddDynamicArguments(m.GetRemoteName()).
-			Run(ctx, &git.RunOpts{
-				Timeout: timeout,
-				Dir:     wikiPath,
-				Stdout:  &stdoutBuilder,
-				Stderr:  &stderrBuilder,
-			}); err != nil {
+
+		// check whether the remote still exists before pruning to avoid prune creating a new remote
```

Comment:
> to avoid prune creating a new remote

What "prune"? "git remote prune" or "git remote update --prune"?

I see neither of them would "create a new remote" in the git manual. Correct me if I was wrong.

Reply:
ae3c338be3